channel/adjudicator.go

```diff
@@ -25,41 +25,76 @@ import (
 )
 
 type (
-	// An Adjudicator represents an adjudicator contract on the blockchain. It has
-	// methods for state registration, progression and withdrawal of channel
-	// funds. A channel state needs to be registered before it can be progressed
-	// or withdrawn. No-App channels skip the force-execution phase and can be
-	// withdrawn directly after the refutation phase has finished. Channels in a
-	// final state can directly be withdrawn after registration.
+
+	// Adjudicator is the interface that groups the Register, Withdraw,
+	// Progress and Subscribe methods.
+	//
+	// An adjudicator represents an adjudicator contract on the blockchain. A
+	// channel state needs to be registered before it can be progressed or
+	// withdrawn. No-App channels skip the force-execution phase and can be
+	// withdrawn directly after the refutation phase has finished. Channels in
+	// a final state can directly be withdrawn after registration.
 	//
 	// Furthermore, an Adjudicator has a method for subscribing to
 	// AdjudicatorEvents. Those events might be triggered by a Register or
 	// Progress call on the adjudicator from any channel participant.
 	Adjudicator interface {
-		// Register should register the given ledger channel state on-chain.
-		// If the channel has locked funds into sub-channels, the corresponding
-		// signed sub-channel states must be provided.
+		Registerer
+		Withdrawer
+		Progresser
+		EventSubscriber
+	}
+
+	// RegisterSubscriber is the interface that groups Register and Subscribe
+	// methods.
+	//
+	// These methods are used to watch for adjudicator events on the blockchain
+	// and register disputes, if state in the adjudicator event is not the
```

Comment:
Disputes are not registered but states, or just write `and dispute`.

Reply:
`the state`

Reply:
Updated in 06a82e12.